### PR TITLE
Refactor help examples to use go templates

### LINKED ISSUE #4654: Help doc for preference is buggy
/kind bug

## What versions of software are you using?
**Operating System:**
Fedora 32

**Output of `odo version`:**
odo v2.1.0 (40a90e175)
Server: https://192.168.42.43:8443
Kubernetes: v1.11.0+d4cacc0

## How did you run odo exactly?
```
odo preference --help
odo preference unset --help
```

## Actual behavior
`odo preference --help`
```
  ...
  ...
  odo preference set %!s(BADINDEX) true

  # Unset a preference value in the global preference
  ...
  ...
  odo preference unset %!s(BADINDEX)
```

`odo preference unset --help`
```
Available Global Parameters:
 ...
 UpdateNotification - Flag to control if an update notification is shown or not (Default: true)
 %!s(BADINDEX)
```
## Expected behavior
There should be no `BADINDEX` error.

### PR DESCRIPTION
**Pull request by valaparthvi**
**What type of PR is this?**
/kind code-refactoring

**What does does this PR do / why we need it**:
This PR refactors the preference examples to use Go templates instead of normal string formatting

**Which issue(s) this PR fixes**:

Fixes #4654 

**PR acceptance criteria**:

- [ ] Unit test 

- [ ] Integration test 

**How to test changes / Special notes to the reviewer**:
```
odo preference --help
odo preference unset --help
```

### CODE REVIEW

File: pkg/odo/cli/preference/set.go [hunk 1]
```diff
@@ -111,3 +104,23 @@ func NewCmdSet(name, fullName string) *cobra.Command {
 	preferenceSetCmd.Flags().BoolVarP(&o.configForceFlag, "force", "f", false, "Don't ask for confirmation, set the preference directly")
 	return preferenceSetCmd
 }
+
+func formatPreferenceExamples(fullName, templater string) string {
```
Comment (valaparthvi) [inline pkg/odo/cli/preference/set.go:108]: I am not very happy about this function placement and perhaps we could write this function in a better way, so please feel free to suggest that :)

**Comment by dharmit** [comment]
@valaparthvi do you plan to take this PR ahead?

**Comment by valaparthvi** [comment]
>@valaparthvi do you plan to take this PR ahead?

Yes, probably this sprint.

File: pkg/odo/cli/preference/set.go [hunk 2]
```diff
@@ -93,10 +84,18 @@ func NewCmdSet(name, fullName string) *cobra.Command {
...
-			preference.UpdateNotificationSetting, preference.NamePrefixSetting,
-			preference.TimeoutSetting, preference.BuildTimeoutSetting, preference.PushTimeoutSetting,
-			preference.ExperimentalSetting, preference.ConsentTelemetrySetting),
+		Example: func(exampleString, fullName string) string {
+			cfg, _ := preference.New()
+			properties := preference.NewPreferenceList(*cfg)
+			for _, property := range properties.Items {
+				value := property.Default
+				if value == "" {
+					value = "smh"
```
Comment (dharmit) [inline pkg/odo/cli/preference/set.go:93]: What's `smh`? I see it show up in example for `odo preference set NamePrefix smh`. Maybe use something simpler?

**Review by dharmit — CHANGES_REQUESTED** [review]
Functionality looks good. One question about `smh`.

File: pkg/odo/cli/preference/set.go (hunk 2 above)
Comment (valaparthvi) [reply pkg/odo/cli/preference/set.go:93]: Yeah, I had my doubts about it :sweat_smile:

**Review by dharmit — APPROVED** [review]

**Pull request merged**